lib/__tests__/plugin.js

```diff
@@ -73,6 +73,46 @@ describe('Middleware', () => {
       })
     })
   })
+
+  describe('when using a function for the authenticated option', () => {
+    describe('with an authenticated component', () => {
+      beforeEach(() => {
+        const comp = { components: [{ options: { authenticated: () => true } }] }
+        context.route.matched.push(comp)
+      })
+
+      it('redirects', async () => {
+        await (Middleware.auth(context))
+
+        expect(context.redirect).toHaveBeenCalled()
+      })
+
+      describe('with an access token', () => {
+        beforeEach(() => {
+          context.store.state.oauth.accessToken = 'i am an access token'
+        })
+
+        it('does nothing', async () => {
+          await (Middleware.auth(context))
+
+          expect(context.redirect).not.toHaveBeenCalled()
+        })
+      })
+    })
+
+    describe('with an unauthenticated component', () => {
+      beforeEach(() => {
+        const comp = { components: [{ options: { authenticated: () => false } }] }
+        context.route.matched.push(comp)
+      })
+
+      it('does not redirect', async () => {
+        await (Middleware.auth(context))
+
+        expect(context.redirect).not.toHaveBeenCalled()
+      })
```

Comment:
worth here checking that the function itself has been called? it could not redirect if something else broke, even if the function was not called...

Reply:
also test that the component definition that is passed as an argument is what we want it to be?

Reply:
will add these tests